services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -48,6 +51,40 @@ def from_botocore_object(obj: ObjectTypeDef) -> "S3MetaData":
         )
 
 
+async def _list_objects_v2_all_items(
+    client: S3Client,
+    bucket: S3BucketName,
+    prefix: str,
+    max_items: int | None = MAX_LIST_OBJECTS_V2_ITEMS,
+) -> list[ObjectTypeDef]:
+    """
+    Scans the given path in prefix and returns information about the files.
+    If `max_items` is None returns all the items in the dataset.
+
+    Keyword Arguments:
+        max_items -- total items to return regardless of pagination
+            (default: {MAX_LIST_OBJECTS_V2_ITEMS})
+    """
+    # NOTE: code was refactored to behave as before. There are risks
+    # of making storage crash, currently the same limits as before
+    # will apply. At most 1000 elements will be returned.
+    # Pagination will be added in the future starting from here.
+
+    pagination_config: PaginatorConfigTypeDef = {}
+    if max_items is not None:
+        pagination_config["MaxItems"] = max_items
+
+    results: deque[ObjectTypeDef] = deque()
```

Comment:
FYI I double checked that deck have no advantage over lists when you append. see for yourself [https://dev.to/wnleao/python-deque-vs-list-time-comparison-5ch4](https://dev.to/wnleao/python-deque-vs-list-time-comparison-5ch4)
on the contrary I guess you lose time by creating a list object again at the end and it also makes it less readable.

Reply:
The article above does not to it justice. Apparently it has the same time complexity. I will trust the wiki https://wiki.python.org/moin/TimeComplexity
Good to know that python's list append is efficient.